### ISSUE TITLE
Different use_constexpr definitions in spdlog and fmt causes compile time error in NVCC (and possibly other compilers)

### ISSUE THREAD

**Issue by ShujianQian** [gabime/spdlog#2856]
# Description
When calling `logger.trace()` in an .cu file compiled using nvcc, I get the following compile time error:
~~~
spdlog/common.h:351:30: error: call to non-‘constexpr’ function ‘const T* fmt::v9::detail::buffer<T>::data() const [with T = char]’
  351 |     return spdlog::string_view_t{buf.data(), buf.size()};
      |                      ~~~~~~~~^~
~~~

# Cause
The [to_string_view](https://github.com/gabime/spdlog/blob/cedfeeb95f3af11df7d3b1e7e0d3b86b334dc23b/include/spdlog/common.h#L349) utility function is declared as [SPDLOG_CONSTEXPR_FUNC](https://github.com/gabime/spdlog/blob/cedfeeb95f3af11df7d3b1e7e0d3b86b334dc23b/include/spdlog/common.h#L72C19-L72C19). The functions calls [fmt::v9::detail::buffer<T>::data()](https://github.com/gabime/spdlog/blob/cedfeeb95f3af11df7d3b1e7e0d3b86b334dc23b/include/spdlog/fmt/bundled/core.h#L911) which is declared as [FMT_CONSTEXPR](https://github.com/gabime/spdlog/blob/cedfeeb95f3af11df7d3b1e7e0d3b86b334dc23b/include/spdlog/fmt/bundled/core.h#L95). However, FMT_CONSTEXPR has different conditions than SPDLOG_CONSTEXPR_FUNC and is defined to be empty for NVCC, leading to the data() function not declared as constexpr and thus causing the error.

# Fix
Adding the [same condition](https://github.com/gabime/spdlog/blob/cedfeeb95f3af11df7d3b1e7e0d3b86b334dc23b/include/spdlog/fmt/bundled/core.h#L95) to the definition of SPDLOG_CONSTEXPR_FUNC fixes the error. However, the two definitions need to be kept in sync in future changes.

Another potential fix is to declare the to_string_view functions as inline instead of constexpr.

**Comment by tt4g** [comment]
PR is welcome.

**Comment by gabime** [comment]
Which version of spdlog?

**Comment by ShujianQian** [comment]
This is introduced in [commit 4f800773](https://github.com/gabime/spdlog/commit/4f800773393d3ebac13c1fcd946a315d4d72bcd9) and is included in v1.12.0 and in the current v1.x branch.

**Comment by cartoonist** [comment]
I came across the same issue using nvcc 11.4 and gcc 10.5.0. PR #2859 fixed it for me (spdlog v1.11.0). Thanks @ShujianQian for the PR!
